src/library/serato/seratofeature.cpp

```diff
@@ -0,0 +1,1043 @@
+            track.location = parseText(data, fieldSize);
+            break;
+        case FieldId::SongTitle:
+            track.title = parseText(data, fieldSize);
+            break;
+        case FieldId::Artist:
+            track.artist = parseText(data, fieldSize);
+            break;
+        case FieldId::Album:
+            track.album = parseText(data, fieldSize);
+            break;
+        case FieldId::Genre:
+            track.genre = parseText(data, fieldSize);
+            break;
+        case FieldId::Length: {
+            bool ok;
+            int duration = parseText(data, fieldSize).toInt(&ok);
+            if (ok) {
+                track.duration = duration;
+            }
+            break;
+        }
+        case FieldId::Bitrate:
+            track.bitrate = parseText(data, fieldSize);
+            break;
+        case FieldId::SampleRate:
+            track.samplerate = parseText(data, fieldSize);
+            break;
+        case FieldId::Bpm: {
+            bool ok;
+            double bpm = parseText(data, fieldSize).toDouble(&ok);
+            if (ok) {
+                track.bpm = bpm;
+            }
+            break;
+        }
+        case FieldId::Comment:
+            track.comment = parseText(data, fieldSize);
+            break;
+        case FieldId::Grouping:
+            track.grouping = parseText(data, fieldSize);
+            break;
+        case FieldId::Label:
+            track.label = parseText(data, fieldSize);
+            break;
+        case FieldId::Year: {
+            bool ok;
+            int year = parseText(data, fieldSize).toInt(&ok);
+            if (ok) {
+                track.year = year;
+            }
+            break;
+        }
+        case FieldId::Key:
+            track.key = parseText(data, fieldSize);
+            break;
+        case FieldId::BeatgridLocked:
+            track.beatgridlocked = parseBoolean(data);
+            break;
+        case FieldId::Missing:
+            track.missing = parseBoolean(data);
+            break;
+        case FieldId::FileTime:
+            track.filetime = parseUInt32(data);
+            break;
+        case FieldId::DateAdded:
+            track.datetimeadded = parseUInt32(data);
+            break;
+        case FieldId::DateAddedText:
+            // Ignore this field, but do not print a debug message
+            break;
+        default:
+            qDebug() << "Ignoring unknown field "
+                     << fieldName
+                     << " ("
+                     << fieldSize
+                     << " bytes).";
+        }
+
+        headerData = buffer.read(8);
+    }
+
+    if (headerData.length() != 0) {
+        qWarning() << "Found "
+                   << headerData.length()
+                   << " extra bytes at end of track definition.";
+        return false;
+    }
+
+    if (track.location.isEmpty()) {
+        qWarning() << "Found track with empty location field.";
+        return false;
```

Comment:
Did you consider to add the track anyway. Maybe Mixxx knows where it is. The user will be happy to persevere all data.
It is finally not a difference if the location is empty, or the file has been moved away.

Reply:
I don't know if is even possible. The location is used as track id in Crates, so an empty location would mean that it can't be in a Crate. I just thought I'd better be safe than sorry. If you want I can remove this check.

Reply:
I this case it makes sense. Maybe you should drop a comment.